test/powershell/Language/Scripting/DeserializedTypeConversion.Tests.ps1

```diff
@@ -0,0 +1,283 @@
+            {
+                get { return name; }
+                set { name = value; }
+            }
+
+            public int Port
+            {
+                get { return port; }
+                set { port = value; }
+            }
+
+            public string ScriptText
+            {
+                get { return scriptText; }
+            }
+        }
+'@
+
+        $type4 = Add-Type -PassThru -TypeDefinition @'
+        public class test4
+        {
+            private string name = "default";
+            private int port = 80;
+            private string scriptText = "1..6";
+
+            public string Name
+            {
+                get { return name; }
+                set { name = value; }
+            }
+
+            public int Port
+            {
+                get { return port; }
+                set { port = value; }
+            }
+
+            internal void Compute()
+            {
+                scriptText = scriptText + " Computed";
+            }
+        }
+'@
+
+        class PSClass1 {
+            [string] $name = "PSClassName1"
+            [int] $port = 80
+            [string] $scriptText = "1..6"
+        }
+
+        class PSClass2 {
+            [string] $name = "default"
+            [int] $port = 80
+            [string] $scriptText = "1..6"
+            PSClass2() {}
+            PSClass2([string] $name1, [int] $port1, [string] $scriptText1)
+            {
+                $this.name = $name1
+                $this.port = $port1
+                $this.scriptText = $scriptText1
+            }
+        }
+
+        function test-1
+        {
+            param(
+                [parameter(position=0, mandatory=1)]
+                [test1] $test
+            )
+
+            $test | Format-List | Out-String
+        }
+
+        function test-2
+        {
+            param(
+                [parameter(position=0, mandatory=1)]
+                [test2] $test
+            )
+
+            $test | Format-List | Out-String
+        }
+
+        function test-3
+        {
+            param(
+                [parameter(position=0, mandatory=1)]
+                [test3] $test
+            )
+
+            $test | Format-List | Out-String
+        }
+
+        function test-4
+        {
+            param(
+                [parameter(position=0, mandatory=1)]
+                [test4] $test
+            )
+
+            $test | Format-List | Out-String
+        }
+
+        function test-PSClass1
+        {
+            param(
+                [parameter(position=0, mandatory=1)]
+                [PSClass1] $test
+            )
+
+            $test | Format-List | Out-String
+        }
+
+        function test-PSClass2
+        {
+            param(
+                [parameter(position=0, mandatory=1)]
+                [PSClass2] $test
+            )
+
+            $test | Format-List | Out-String
+        }
+
+        $t1 = new-object test1 -Property @{name="TestName1";port=80;scriptText="1..5"}
```

Comment:
fixed

---
In reply to: [78838934](https://github.com/Powershell/Powershell/pull/2118#discussion_r78838934) [](ancestors = 78838934)